### ISSUE TITLE
[Docs]: Missing documentation on relation resolvers in a service

### ISSUE THREAD

**Issue by dac09** [redwoodjs/redwood#6474]
### Summary and description

We don't really talk about how the relation resolvers generated at the bottom of a service file work. 

The only mention is here: https://redwoodjs.com/docs/graphql#redwoods-resolver-args - which is useful, but not enough IMO.

What do I mean by relation resolvers? 

```
// services/posts.ts

export const post: QueryResolvers['post'] = /**/
export const createPost: MutationResolvers['createPost'] =  /**/

// 👇 these are your "relation" or field resolvers
- export const Post: PostResolvers = {
+ export const Post: PostRelationResolvers = {
  author: (_obj, gqlArgs) => /**/
}
```

These resolvers are invoked when a "relation" on the Post model is invoked, in this case if the query requested `Post.author.name` for example.

**Things we should cover:**
- how does a query/mutation map to a service function?
- what is the order of preference i.e. if you define a relation resolver for a field, it will be invoked first
- how to optimise queries either:
a) use prisma includes in the main service function + check if already included in root) 
b) use dataloaders

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by cannikin** [comment]
I've started on this in the tutorial here: https://github.com/redwoodjs/redwood/pull/6465

**Comment by dac09** [comment]
@cannikin is it worth adding a section the docs/reference (in the services section) and not just the tutorial?

It honestly is one of the most confusing things in services - happy to give it a go if you want to focus on the tutorial piece.

**Comment by cannikin** [comment]
Yep, I had planned to add a more in-depth section in the Services docs themselves. Feel free to add something if you need a break from Typescript for a while! :)

**Comment by cannikin** [comment]
This probably shouldn't be closed—this was referring to a more in-depth description of relation resolvers, while the intro in the tutorial is pretty minor.

**Comment by cannikin** [comment]
I assigned this over to you @dac09 but let me know if you change your mind!
